provisioning/baremetal_config.go

```diff
@@ -61,9 +64,6 @@ func getDHCPRange(config *metal3iov1alpha1.ProvisioningSpec) *string {
 }
 
 func getProvisioningIPCIDR(config *metal3iov1alpha1.ProvisioningSpec) *string {
-	if config.ProvisioningNetwork == metal3iov1alpha1.ProvisioningNetworkDisabled {
-		return nil
-	}
 	if config.ProvisioningNetworkCIDR != "" && config.ProvisioningIP != "" {
```

Comment:
I wonder if we need to split the logic around provisioning CIDR and provisioning IP. Despite very similar name, the latter is generically useful while the former is DHCP-specific.

Shameless plug: when writing IrSO, I actually split the two option groups entirely: https://github.com/metal3-io/ironic-standalone-operator/blob/main/api/v1alpha1/ironic_types.go#L61-L161